src/clients/KubectlClient.ts

```diff
@@ -101,10 +100,73 @@ export class KubectlClient implements IClient {
         }
     }
 
+    public async getContainersList(podName: string): Promise<string[]> {
+        if (!podName) {
+            this._logger.error(TelemetryEvent.KubectlClient_GetPodNameError, new Error(`Pod name is null`));
+            return null;
+        }
+        try {
+            // adding these decent amount of known sidecars to filter out the sidecars from the list
+            const knownSideCars: string[] = ['linkerd-proxy', 'linkerd-init', 'istio-proxy', 'darpd', 'jaeger-agent', 'nginx-proxy'];
+            const kubeconfigPath: string = await this._accountContextManager.getKubeconfigPathAsync();
+            const args: string[] = [`get`, `pod`, podName, `-o`, `jsonpath="{.spec['containers','initContainers'][*].name}"`];
+            const kubectlOutput: string = await this.runKubectlCommandAsync(args, kubeconfigPath);
+            if (kubectlOutput) {
+                // replace the double quotes and split the string by space ex: "linkerd-proxy stats-api linkerd-init" to [linkerd-proxy, stats-api, linkerd-init]
+                // and filter out the known sidecars from the list ex: [stats-api]
+                return kubectlOutput.replace('"', '').replace('"', '').split(' ')
+                    .filter(s => !knownSideCars.find(knownSideCar => knownSideCar == s));
+            } else {
+                this._logger.error(TelemetryEvent.KubectlClient_GetContainerListError, new Error(`container list is null`));
+                return null;
+            }
+        } catch (error) {
+            this._logger.error(TelemetryEvent.KubectlClient_GetContainerListError, error);
+            return null;
+        }
+
+    }
+
+    public async getPodName(serviceName: string): Promise<string> {
+        try {
+            const kubeconfigPath: string = await this._accountContextManager.getKubeconfigPathAsync();
+            // find the pod ipaddressList for the selected service
+            const args: string[] = [`get`, `ep`, serviceName, `-o`, `jsonpath='{.subsets[*].addresses[*].ip}'`];
```

Comment:
Again, need namespace?

Reply:
I am refactoring all this with javascript client for k8s. this will be replaced